lib/kafka-worker.rb

```diff
@@ -38,16 +48,20 @@ def run
           if message.topic == handler.topic
             handler_obj = handler.new
             handler_obj.logger = @logger
-            begin
-              handler_obj.handle(message)
-            rescue Exception => err
-              handler_obj.on_error(message, err)
+            (0..3).each do
+              begin
+                handler_obj.handle(message)
+                break
```

Comment:
Better use Ruby's `retry` instead of a loop with break - see for instance http://blog.mirthlab.com/2012/05/25/cleanly-retrying-blocks-of-code-after-an-exception-in-ruby/.

Even better would be if we could push the event back into kafka to be delivered again later, but seems that's quite difficult to achieve. I found http://pragmatists.pl/blog/2017/03/retrying-consumer-architecture-in-the-apache-kafka/ on this.